### PR TITLE
chore: add GPL3 License with EXCEPTIONS

### PR DESCRIPTION
**Pull request by alandtse**
This adds a GPL3 or later License with exceptions that are conducive to game modding. Specifically, the exceptions:
1. Allow the linking with Starfield and SFSE without requiring the distribution of their source code or that they be treated as GPL too. Starfield is proprietary and SFSE is still considering license choices.
2. Allow the linking of other libraries without requiring that those libraries be distributed as GPL, but the library's source code must be provided. Please note, those libraries may be non-free but free alternatives may be possible given the access to the source code.

Otherwise, the terms of GPL-3-or-later apply. See the FSF FAQ at: https://www.gnu.org/licenses/gpl-faq.en.html

### CODE REVIEW

**Review by ThirdEyeSqueegee — APPROVED** [review]

**Comment by gottyduke** [comment]
so I should nuke the original MIT license from CLib code? #3

**Comment by alandtse** [comment]
I recommend leaving this open for a little so people can comment/ask questions. If we have any PRs pending (#3) we want in, just ask them to approve this PR before merging. @gottyduke (approve if you're good with this license).

> so I should nuke the original MIT license from CLib code? #3
No, we'll have to keep it. Probably move it to a subdirectory called LICENSES and mention it in the README as covering the original bits of commonlib we took.

**Review by gottyduke — APPROVED** [review]

**Comment by ianpatt** [comment]
GPL3 works for me here. I strongly feel that if someone wants to piggyback on existing work, they have to give back as well. I can commit that whatever license SFSE ends up under, the source code will be available or better. I want people to be able to use the code, but need to prevent stupid people from doing stupid things. (as I've seen with all the script extender projects in the past)

We're in a very different place with Starfield compared with the previous games. Usually it's me and Stephen or Expired poking at things right at release time, and then a major project like himika's library or the various CommonLib follows later. Now we've got a lot of people all interested at release time.

**Review by Vermunds — APPROVED** [review]

**Pull request merged**
